$OpenBSD: patch-lib_usual_tls_tls_compat_h,v 1.3 2021/10/31 16:59:13 tb Exp $

Neuter compat shims that break with opaque structs in ssl.h and x509.h.

Index: lib/usual/tls/tls_compat.h
--- lib/usual/tls/tls_compat.h.orig
+++ lib/usual/tls/tls_compat.h
@@ -28,7 +28,6 @@
 
 #ifndef LIBRESSL_VERSION_NUMBER
 #define USE_LIBSSL_INTERNALS
-#endif
 
 #define NID_kx_ecdhe (-90)
 #define NID_kx_dhe (-91)
@@ -79,7 +78,7 @@ static inline X509_OBJECT *X509_STORE_CTX_get_obj_by_s
 	return NULL;
 }
 
-
+#endif
 #endif
 
 /* ecdh_auto is broken - ignores main EC key */
